src/util/pointer_predicates.cpp

```diff
@@ -171,17 +171,16 @@ exprt dynamic_object_upper_bound(
   {
     op=ID_gt;
 
-    if(ns.follow(object_offset.type())!=
-       ns.follow(access_size.type()))
-      object_offset.make_typecast(access_size.type());
-    sum=plus_exprt(object_offset, access_size);
+    const auto object_offset_casted =
+      typecast_exprt::conditional_cast(object_offset, access_size.type());
```

Comment:
I'm not entirely happy with this happening here - on the one hand because it's already being done in https://github.com/diffblue/cbmc/pull/3737/commits/8f52d813cb46a2bc6da01b41b21762ebca4a46ba, and on the other hand because it does only one part of the work (it doesn't remove `ns` from the parameter list).